### PR TITLE
cody: scroll to view on new chat messages

### PR DESCRIPTION
**Pull request by abeatrix**
RE [slack thread
](https://sourcegraph.slack.com/archives/C04NPH6SZMW/p1678900652268589)

1. Add: auto scroll to the bottom of the screen on new chat messages ([Loom](https://www.loom.com/share/5bf6b5d9bef6446291d5a8ef480f7776))

cody

2. Replace `arrow-up` icon with `arrow-down`  on expanded context messages.

Also updated the default font color for bot messages

<table>
<tr>
<td>Before</td>
<td>After</td>
</tr>
<tr>
<td>
image
</td>
<td>
image
</td>
</tr>
</table>

## Test plan

launch Cody in VS Code Debug mode:
- Run `pnpm i` at the root of this repository
- Select `Launch Cody Extension` from the dropdown menu in the `RUN AND DEBUG` sidebar in VS Code

## App preview:

- [Web](https://sg-web-bee-cody-chat-scroll.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/cody/vscode-codegen/webviews/Chat.tsx [hunk 1]
```diff
@@ -160,7 +170,7 @@ const ContextFiles: React.FunctionComponent<{ contextFiles: string[] }> = ({ con
         return (
             <p className="context-files-expanded">
                 <span className="context-files-toggle-icon" onClick={() => setIsExpanded(false)}>
```
Comment (dominiccooney) [inline client/cody/vscode-codegen/webviews/Chat.tsx:172]: Idea for later, <details open><summary>there's a widget built into HTML for this.</summary>It's the `<details>` and `<summary>` elements. Don't worry, it is very styleable.</details>

File: client/cody/vscode-codegen/webviews/Chat.tsx [hunk 2]
```diff
@@ -63,6 +64,14 @@ export const Chat: React.FunctionComponent<React.PropsWithChildren<ChatboxProps>
 
     const bubbleClassName = (speaker: string): string => (speaker === 'you' ? 'human' : 'bot')
 
+    const scrollToBottom = () => {
+        chatboxRef.current?.scrollIntoView({ behavior: 'smooth' })
```
Comment (dominiccooney) [inline client/cody/vscode-codegen/webviews/Chat.tsx:68]: You could `?.` the call so we don't throw when `current` isn't set, if that can happen.

```suggestion
        chatboxRef.current?.scrollIntoView?.({ behavior: 'smooth' })
```

**Review by dominiccooney — APPROVED** [review]
Looks good. Some feedback inline.

Yay for increased contrast on important content.

Idea for later, we could also make a "cody is typing" or "new messages" floating widget which appears if the scroll is not stuck to the bottom, which lets you scroll to the bottom.

**Comment by abeatrix** [comment]
Thanks @dominiccooney !

**Pull request merged**